src/main/java/net/dv8tion/jda/core/entities/impl/ReceivedMessage.java

```diff
@@ -0,0 +1,822 @@
+                    if (isFromType(ChannelType.TEXT)) // role lookup is faster if its in the same guild (no global map)
+                        role = getGuild().getRoleById(id);
+                    if (role == null)
+                        role = getJDA().getRoleById(id);
+                    if (role != null)
+                        roleMentions.add(role);
+                }
+                catch (NumberFormatException ignored) {}
+            }
+            roleMentions = Collections.unmodifiableList(roleMentions);
+        }
+
+        return roleMentions;
+    }
+
+    @Override
+    public List<Member> getMentionedMembers(Guild guild)
+    {
+        Checks.notNull(guild, "Guild");
+        List<User> mentionedUsers = getMentionedUsers();
+        List<Member> members = new ArrayList<>();
+        for (User user : mentionedUsers)
+        {
+            Member member = guild.getMember(user);
+            if (member != null)
+                members.add(member);
+        }
+
+        return Collections.unmodifiableList(members);
+    }
+
+    @Override
+    public List<Member> getMentionedMembers()
+    {
+        if (isFromType(ChannelType.TEXT))
+            return getMentionedMembers(getGuild());
+        else
+            throw new IllegalStateException("You must specify a Guild for Messages which are not sent from a TextChannel!");
+    }
+
+    @Override
+    public List<IMentionable> getMentions(MentionType... types)
+    {
+        if (types == null || types.length == 0)
+            return getMentions(MentionType.values());
+        List<IMentionable> mentions = new ArrayList<>();
+        // boolean duplicate checks
+        // not using Set because channel and role might have the same ID
+        boolean channel = false;
+        boolean role = false;
+        boolean user = false;
+        boolean emote = false;
+        for (MentionType type : types)
+        {
+            switch (type)
+            {
+                case EVERYONE:
+                case HERE:
+                default: continue;
+                case CHANNEL:
+                    if (!channel)
+                        mentions.addAll(getMentionedChannels());
+                    channel = true;
+                    break;
+                case USER:
+                    if (!user)
+                        mentions.addAll(getMentionedUsers());
+                    user = true;
+                    break;
+                case ROLE:
+                    if (!role)
+                        mentions.addAll(getMentionedRoles());
+                    role = true;
+                    break;
+                case EMOTE:
+                    if (!emote)
+                        mentions.addAll(getEmotes());
+                    emote = true;
+            }
+        }
+        return Arrays.asList(mentions.toArray(new IMentionable[mentions.size()]));
+    }
+
+    @Override
+    public boolean isMentioned(IMentionable mentionable, MentionType... types)
```

Comment:
Can't you get the `MentionType` from the `IMentionable`?

Reply:
The mention types are used in order to check if a user was mentioned by role for instance

Reply:
Oh, that makes sense I guess.